content/docs/concepts/remote_write_spec_2_0.md

```diff
@@ -0,0 +1,394 @@
+
+* `User-Agent: <name & version of the sender>`
+* `X-Prometheus-Remote-Write-Version: <remote write specificiation version the sender follows>`
+
+  Sender SHOULD use `X-Prometheus-Remote-Write-Version: 0.1.0` for backward compatibility, when using 1.0 proto message.
+
+Sender MAY allow users to send custom HTTP headers; they MUST NOT allow users to configure them in such a way as to send reserved headers.
+
+<!---
+TODO(bwplotka): Kind of repeated statement, feedback welcome how to structure it.
+-->
+The remote write request in the body of the HTTP POST MUST be compressed with [Google’s Snappy](https://github.com/google/snappy). The block format MUST be used -- the framed format MUST NOT be used. The remote write request MUST be encoded using Google Protobuf 3, and MUST use either of the schemas defined above. 
+
+#### `io.prometheus.write.v2.Request` Proto Schema
+
+<!---
+TODO(bwplotka): Move link to the one on Prometheus main or even buf. 
+-->
+The source of truth is [here](https://github.com/prometheus/prometheus/blob/remote-write-2.0/prompb/io/prometheus/write/v2/types.proto#L32). The `gogo` dependency and options CAN be ignored. They are not part of the specification as they don't impact the serialized format.
+
+The simplified version of the new `io.prometheus.write.v2.Request` is presented below.
+
+```
+// Request represents a request to write the given timeseries to a remote destination.
+message Request {
+  // symbols contains a de-duplicated array of string elements used for various
+  // items in a Request message, like labels and metadata items. To decode
+  // each of those items, referenced, by "ref(s)" suffix, you need to lookup the
+  // actual string by index from symbols array. The order of strings is up to
+  // the client, server should not assume any particular encoding.
+  repeated string symbols = 1;
+  // timeseries represents an array of distinct series with 0 or more samples.
+  repeated TimeSeries timeseries = 2;
+}
+
+// TimeSeries represents a single series.
+message TimeSeries {
+  // labels_refs is a list of label name-value pair references, encoded
+  // as indices to the Request.symbols array. This list's length is always
+  // a multiple of two, and the underlying labels should be sorted.
+  //
+  // Note that there might be multiple TimeSeries objects in the same
+  // Requests with the same labels e.g. for different exemplars, metadata
+  // or created timestamp.
+  repeated uint32 labels_refs = 1;
+
+  // Timeseries messages can either specify samples or (native) histogram samples
+  // (histogram field), but not both. For typical clients (real-time metric
+  // streaming), in healthy cases, there will be only one sample or histogram.
```

Comment:
should we include a note `exporting both a regular metric sample and histogram sample for the exact same metric name and label set should not be possible, and if it occurs within a client library or remote write implementation it should be considered an error`?